borg/_hashindex.c

```diff
@@ -220,6 +230,19 @@ int shrink_size(int current){
     return hash_sizes[i];
 }
 
+int
+count_empty(HashIndex *index)
+{   /* count empty (never used) buckets. this does NOT include deleted buckets (tombstones).
+     * TODO: if we ever change HashHeader, save the count there so we do not need this function.
```

Comment:
for future reference: mmaping the table thus does not make sense with this code here.